index.js

```diff
@@ -325,6 +332,76 @@ class GithubScm extends Scm {
             url: branchUrl
         }));
     }
+
+    /**
+     * Given a SCM webhook payload & its associated headers, aggregate the
+     * necessary data to execute a Screwdriver job with.
+     * @method parseHook
+     * @param  {Object}  webhookPayload  The webhook payload received from the
+     *                                   SCM service.
+     * @param  {Object}  payloadHeaders  The request headers associated with the
+     *                                   webhook payload
+     * @return {Object}                  A key-map of data related to the received
+     *                                   payload
+     */
+    parseHook(webhookPayload, payloadHeaders) {
+        const type = payloadHeaders['x-github-event'];
+        const checkoutUrl = hoek.reach(webhookPayload, 'repository.ssh_url');
+
+        switch (type) {
+        case 'pull_request': {
+            let action = hoek.reach(webhookPayload, 'action');
+            const prNum = hoek.reach(webhookPayload, 'pull_request.number');
+
+            if (!SUPPORTED_PR_ACTIONS.includes(action)) {
+                action = 'closed';
+            }
+
+            return {
+                action,
+                branch: hoek.reach(webhookPayload, 'pull_request.base.ref'),
+                checkoutUrl,
+                prNum,
+                prRef: `${checkoutUrl}#pull/${prNum}/merge`,
+                sha: hoek.reach(webhookPayload, 'pull_request.head.sha'),
+                type: 'pr',
+                username: hoek.reach(webhookPayload, 'pull_request.user.login')
+            };
+        }
+        case 'push':
+            return {
+                action: 'opened',
+                branch: hoek.reach(webhookPayload, 'ref').replace(/^refs\/heads\//, ''),
+                checkoutUrl: hoek.reach(webhookPayload, 'repository.ssh_url'),
+                sha: hoek.reach(webhookPayload, 'after'),
+                type: 'repo',
+                username: hoek.reach(webhookPayload, 'sender.login')
+            };
+        default:
+            throw new Error(`Event ${type} not supported`);
+        }
+    }
+
+    /**
+     * Parses a SCM URL into a screwdriver-representable ID
+     *
+     * 'token' is required, since it is necessary to lookup the SCM ID by
+     * communicating with said SCM service.
+     * @method parseUrl
+     * @param  {Object} config         Config object
+     * @param  {String} config.scmUrl  The scmUrl to parse
+     * @param  {String} config.token   The token used to authenticate to the SCM service
+     * @return {Promise}               Resolves to an ID of 'serviceName:repoId:branchName'
+     */
+    parseUrl(config) {
+        const scmInfo = getInfo(config.scmUrl);
```

Comment:
`scmUrl` is now `checkoutUrl`
https://github.com/screwdriver-cd/scm-github/pull/14/files#diff-04c6e90faac2675aa89e2176d2eec7d8R58